apiserver/migrationtarget/migrationtarget.go

```diff
@@ -223,3 +224,70 @@ func (api *API) AdoptResources(args params.AdoptResourcesArgs) error {
 	}
 	return errors.Trace(env.AdoptResources(model.ControllerUUID(), args.SourceControllerVersion))
 }
+
+// CheckMachines compares the machines in state with the ones reported
+// by the provider and reports any discrepancies.
+func (api *API) CheckMachines(args params.ModelArgs) (params.ErrorResults, error) {
+	var empty params.ErrorResults
+	model, err := api.getModel(args.ModelTag)
+	if err != nil {
+		return empty, errors.Trace(err)
+	}
+	st, release, err := api.pool.Get(model.UUID())
+	if err != nil {
+		return empty, errors.Trace(err)
+	}
+	defer release()
+
+	machines, err := st.AllMachines()
+	if err != nil {
+		return empty, errors.Trace(err)
+	}
+	machinesByInstance := make(map[string]string)
+	for _, machine := range machines {
+		if machine.IsContainer() {
+			// Containers don't correspond to instances at the
+			// provider level.
+			continue
+		}
```

Comment:
Probably worthwhile also skipping manual machines.  There is a method "IsManual", that returns (bool, error).